yield_plugin/src/yield_plugin.cpp

```diff
@@ -37,72 +37,344 @@ using oss = std::ostringstream;
+    boost::geometry::model::linestring<lanelet::BasicPoint2d> traj2;
+    for (auto tpp1:trajectory1)
+    {
+      boost::geometry::append(traj1, tpp1);
+    }
+
+    for (auto tpp2:trajectory2)
+    {
+      boost::geometry::append(traj2, tpp2);
+    }
+
+    std::vector<lanelet::BasicPoint2d> intersection_points;
+    boost::geometry::intersection(traj1, traj2, intersection_points);
+
+    return intersection_points;
+  }
+
+  std::vector<lanelet::BasicPoint2d> YieldPlugin::convert_eceftrajectory_to_mappoints(const cav_msgs::Trajectory& ecef_trajectory, const geometry_msgs::TransformStamped& tf) const
+  {
+    cav_msgs::TrajectoryPlan trajectory_plan;
+    std::vector<lanelet::BasicPoint2d> map_points;
+    lanelet::BasicPoint2d first_point;
+
+    first_point.x() = ecef_trajectory.location.ecef_x * tf.transform.translation.x;
+    first_point.y() = ecef_trajectory.location.ecef_y * tf.transform.translation.y;
+
+    map_points.push_back(first_point);
+
+    for (size_t i = 0; i<ecef_trajectory.offsets.size(); i++)
+    {
+      lanelet::BasicPoint2d offset_point;
+      double offset_x = ecef_trajectory.offsets[i].offset_x + ecef_trajectory.location.ecef_x;
+      double offset_y = ecef_trajectory.offsets[i].offset_y + ecef_trajectory.location.ecef_y;
+      offset_point.x() = offset_x * tf.transform.translation.x;
+      offset_point.y() = offset_y * tf.transform.translation.y;
+      map_points.push_back(offset_point);
+    }
+    
+    return map_points;
+  }
+
+  cav_msgs::MobilityResponse YieldPlugin::compose_mobility_response(const std::string& resp_recipient_id, const std::string& req_plan_id) const
+  {
+    cav_msgs::MobilityResponse out_mobility_response;
+    out_mobility_response.header.sender_id = config_.vehicle_id;
+    out_mobility_response.header.recipient_id = resp_recipient_id;
+    out_mobility_response.header.sender_bsm_id = host_bsm_id_;
+    out_mobility_response.header.plan_id = req_plan_id;
+    out_mobility_response.header.timestamp = ros::Time::now().toSec()*1000;
+
+
+    if (config_.always_accept_mobility_request)
+    {
+      out_mobility_response.is_accepted = true;
+    }
+    else out_mobility_response.is_accepted = false;
+    
+    return out_mobility_response;
+  }
+
+
+  void YieldPlugin::mobilityrequest_cb(const cav_msgs::MobilityRequestConstPtr& msg)
+  {
+    cav_msgs::MobilityRequest incoming_request = *msg;
+    cav_msgs::LaneChangeStatus lc_status_msg;
+    if (incoming_request.strategy == "carma/cooperative-lane-change")
+    {
+      if (incoming_request.plan_type.type == cav_msgs::PlanType::CHANGE_LANE_LEFT || incoming_request.plan_type.type == cav_msgs::PlanType::CHANGE_LANE_RIGHT)
+      {
+        ROS_DEBUG_STREAM("Cooperative Lane Change Request Received");
+        lc_status_msg.status = cav_msgs::LaneChangeStatus::REQUEST_RECEIVED;
+        lc_status_msg.description = "Cooperative Lane Change Request Received";
+        if (incoming_request.header.recipient_id == config_.vehicle_id)
```

Comment:
just an extra check to make sure it is receiving and processing a request that is meant for host vehicle.

Reply:
Sounds good, I believe we had also defined an empty-string value here as being the "broadcast recipient", so it'd be good to check for that as well as the merging vehicle may choose to broadcast this message rather than uni-cast it. Unless the CLC plugin as developed by @adev4a is guaranteed to target a specific vehicle with its mobility request, in which case no change is needed. We just need to make sure these are aligned.